fe1-web/src/core/network/ingestion/ExtendedMessage.ts

```diff
@@ -1,28 +1,34 @@
 import {
   Base64UrlData,
+  Base64UrlDataState,
   Channel,
   getLaoIdFromChannel,
   Hash,
+  HashState,
+  ProtocolError,
   PublicKey,
+  PublicKeyState,
   Signature,
+  SignatureState,
   Timestamp,
+  TimestampState,
   WitnessSignature,
   WitnessSignatureState,
 } from 'core/objects';
 
 import { Message, ProcessableMessage } from '../jsonrpc/messages';
 
 export interface ExtendedMessageState {
-  receivedAt: number;
+  receivedAt: TimestampState;
   receivedFrom: string;
-  processedAt?: number;
+  processedAt?: TimestampState;
   channel: Channel;
-  laoId: string;
+  laoId: HashState;
 
-  data: string;
-  sender: string;
-  signature: string;
-  message_id: string;
+  data: Base64UrlDataState;
+  sender: PublicKeyState;
+  signature: SignatureState;
+  message_id: HashState;
```

Comment:
I added type aliases such as `type TimestampState = number` to make it easier to change and also to make the respective "unserialized" types more visible in these type definitions.

Reply:
I agree that it is not clear what the underlying type is but this is what partially motivated the change as it is no longer necessary to know how the different objects are actually stored. The only 2 functions that you need to know that are `fromState` and `toState`. As long as the underlying storage type is not changed it is possible to abstract and completely ignore the actual value of `XXState`. When writing code it should suffice that there is some storage type and I have access to a function that creates an instance from this type (`fromState`) and a function that transforms an instance to this type (`toState`).

My hope is that this also reduces the likelihood of people using the `XXState` anywhere in the application again (as it was the case for `Hash / string`). To me this more clearly indicates that it should only be used for storage, i.e. it is not the "proper" type.

But this of course is a very subjective interpretation and if this is not what other people think when looking at it we should think of a better solution / revert this change!

Reply:
Actually, I find your argument convincing 👍